### PR TITLE
Infra/modifiers perf improvement

### PR DESCRIPTION
**Pull request by lidord-wix**
## Description
Update some modifiers extraction methods to improve performance. 
I checked a specific session (on the private demo app, iOS) before and after these changes. the results are in ms: 
```
before: {
  "backgroundColor": 291.95974069833755,
  "color": 1362.02895539999,
  "typography": 878.3454459905624
},
after: {
  "backgroundColor": 54.971199452877045,
  "color": 97.36799198389053,
  "typography": 85.1839787364006
},
```
Also, I added a function to update the modifiers' colors that should be called after loading some colors/scheme.
(and I call it on private in `preset.config.ts`)

## Changelog
Improve modifiers' performance

### CODE REVIEW

File: src/commons/modifiers.ts [hunk 1]
```diff
@@ -100,30 +103,34 @@ export type ContainerModifiers = AlignmentModifiers &
   BorderRadiusModifiers &
   BackgroundColorModifier;
 
+export function updateColorModifiers() {
```
Comment (ethanshar) [inline src/commons/modifiers.ts:106]: If you'll think about it, maybe it's better that the Colors class will maintain this logic and provide a cached list of colors keys. 
By moving this logic to the Colors class, you can also update the cached list whenever loadColors is being called - so you don't need to worry about it as a consumer. 
WDYT?

Comment (lidord-wix) [reply src/commons/modifiers.ts:106]: I wondered about it.. you can see my note in this PR description: https://github.com/wix-private/wix-react-native-ui-lib/pull/2473

Comment (ethanshar) [reply src/commons/modifiers.ts:106]: Oh ok, I didn't see the private PR yet. 
I think we should call and keep it in the Colors class, what are the cons for doing that?

Comment (lidord-wix) [reply src/commons/modifiers.ts:106]: The only disadvantage I can think of is that it'll cause an overhead in the `loadColors` method in a case that the user calls it many times in his app..  but I guess it's a rare case and we can ignore it
I'll update it :)

**Review by ethanshar — APPROVED** [review]

**Pull request merged**